### ISSUE TITLE
[a11y]: Keyboard Trap when Tabbing using Tooltip

### ISSUE THREAD

**Issue by wfdcfs** [carbon-design-system/carbon#9683]
**Package**  
carbon-components-react

**Browser**  
Firefox

**Operating System**  
MacOS

**Package version**  
v7.43.0

**Automated testing tool and ruleset**  
No. Uncovered via Manual testing

**Assistive technology**  

**Description**  
Expect to be able to use keyboard to tab through each element.

**Failure mechanism**
The screenshot below has 3 'Tags' ('pet', 'store' and 'user') that use Carbon tooltip component in the tags.

If the tooltip has a link in (see screenshot), then once the tooltip gets focus and displays the tooltip, then you are stuck in a keyboard accessibility trap.  You cannot tab forward to next item, or tab + shift will not allow you to go back to previous item.  Pressing 'Esc' key will close the tooltip, and will allow you to use tab + shift to go back to previous item, but not 'tab' to move to next item.  This behaviour is only seen when a link is contained in the tooltip:

![image](https://user-images.githubusercontent.com/25774547/133655043-b2c3c66f-2e97-40b5-80a2-c037077ae59a.png)

If I manually click on the 'store' tag, the tooltip expands, and because there is no link contained in this tooltip, there is no issue with using 'tab' to move to the next item ('user' tag in this case).
![image](https://user-images.githubusercontent.com/25774547/133655655-0cf3eec3-705f-41fd-a7d4-aff5ddd959dd.png)

Again, 'user' tooltip is a keyboard trap as there is a link in this item:
![image](https://user-images.githubusercontent.com/25774547/133655623-01addeba-c515-4fee-8369-a9a8b2524158.png)

Code snippet (removed content that is not relevant including classNames)
```
import { Link, Tooltip } from 'carbon-components-react';
...
<Tooltip triggerText={tag} tabIndex={0} direction="bottom">    
       <...another React component here>
       <div>
            <Link href="href here" target="_blank" rel="noopener noreferrer">Text here</Link>
       </div>
</Tooltip>
```
The focus appears to be stuck on this html snippet (based on 'pet' tag from above image):
```
<a href="http://swagger.io" rel="noopener noreferrer" target="_blank">
      "Find out more: "
       "http://swagger.io"
</a>
```

**Comment by andreancardona** [comment]
@wfdcfs hello! 

Thank you for opening this ticket. I'm happy to look into this more. 

1. Do you mind adding this to a code-sandbox so that we can interact with the issue? https://codesandbox.io/s/github/carbon-design-system/carbon/tree/main/packages/react/examples/codesandbox

Thank you so much!

**Comment by wfdcfs** [comment]
I have created a code-sandbox here: https://codesandbox.io/s/unruffled-violet-tcv6f?file=/src/index.js

If you click on the 'Dummy 1' button, and then hit 'tab' to move to the "Tooltip" React component, you will be brought to the "pet" tooltip, which will show the link.  However, you cannot use 'tab' now to get to the next Tooltip ("store" in the sandbox example).  
If you manually click on 'store' tooltip, because there is no link contained in the tooltip, you can now use 'tab' to move to the next tooltip ('user' in the sandbox example).  Once on the 'user' tooltip, as this has a link, you cannot use 'tab' now to get to the next item in the UI (the 'Dummy 2' button in the sandbox example).  
Note: Buttons 'Dummy 1' and 'Dummy 2' were added just to have html elements before and after the Tooltips.

My expectation is that you should be able to tab as follows: 'Dummy 1' -> Pet Tooltip -> store Tooltip -> use Tooltip -> 'Dummy 2' button

**Comment by chrisdudley** [comment]
@andreancardona is there any update on this please ?

**Comment by andreancardona** [comment]
@chrisdudley I have removed the labels and this ticket will now be added to our backlog :)

**Comment by abbeyhrt** [comment]
Hey @wfdcfs! 👋  This is the intended behavior for Tooltips with interactive content inside of them. Since there is interactive content, we want tooltip to persist until it is dismissed through the escape key instead of being just another tab stop. This approach will change in v11 as Tooltips will no longer be able to have interactive content inside of them (we have a new component incoming to handle that case) but for now, this is how tooltips with a link inside should behave.  Let me know if you have any questions!

**Comment by joshblack** [comment]
Going to close this out as this is something that we're unable to address in v10 and has been addressed in v11. Let me know if you have any questions!
